src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -706,6 +706,18 @@ private static void ReportBinaryOperatorError(ExpressionSyntax node, DiagnosticB
                     Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnDefault, node, operatorToken.Text);
                     return;
                 }
+                else if (leftDefault && right.Type is TypeParameterSymbol)
+                {
+                    Debug.Assert(!right.Type.IsReferenceType);
+                    Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnUnconstrainedDefault, node, operatorToken.Text, right.Display);
+                    return;
+                }
+                else if (rightDefault && left.Type is TypeParameterSymbol)
+                {
+                    Debug.Assert(!left.Type.IsReferenceType);
+                    Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnUnconstrainedDefault, node, operatorToken.Text, left.Display);
```

Comment:
>left.Display [](start = 117, length = 12)

```left.Type```?

Reply:
You mean because we don't need to display "default"?
I'm using the same thing as in the original diagnostic, except just one of them. Seems fine.

Reply:
> I'm using the same thing as in the original diagnostic, except just one of them. Seems fine.

Original code path has to deal with situations when the argument is not always a type, an expression could have no type, for example, and something else is used to fill the spot. In our case we have a type and **want** a type and really shouldn't depend on what Display for an expression might do. 

---
In reply to: [364883115](https://github.com/dotnet/roslyn/pull/40801#discussion_r364883115) [](ancestors = 364883115)